wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession+UserSession.swift

```diff
@@ -294,6 +294,29 @@ extension ZMUserSession: UserSession {
         let featureRepository = FeatureRepository(context: syncContext)
         return GetMLSFeatureUseCase(featureRepository: featureRepository)
     }
+
+    @MainActor
+    public func fetchSelfConversationMLSGroupID() async -> MLSGroupID? {
+        return await syncContext.perform {
+            return ZMConversation.fetchSelfMLSConversation(in: self.syncContext)?.mlsGroupID
+        }
+    }
+
+    @MainActor
+    public func e2eIdentityUpdateCertificateUpdateStatus() async -> E2EIdentityCertificateUpdateStatusProtocol? {
+        guard let mlsGroupID = await self.fetchSelfConversationMLSGroupID(), let selfMLSClientID else {
+            return nil
+        }
+        var updateE2EICertificate: E2EIdentityCertificateUpdateStatusProtocol = E2EIdentityCertificateUpdateStatusUseCase(
+            e2eCertificateForCurrentClient: getE2eIdentityCertificates,
+            gracePeriod: Double(e2eiFeature.config.verificationExpiration),
```

Comment:
`e2eiFeature` must be accessed from the view context's queue.

Reply:
done. Good catch